### DISCUSSION TITLE
Model Data Source being another Model

### DISCUSSION THREAD

**Discussion by rhains** [TabularEditor/TabularEditor3#1006] [🙋♀️TE3 Q&A] [answered]
I am trying to create a Legacy New Provider Data source that connects to a Analysis Service Model in Premium services.

My connection string is 
provider=MSOLAP.8;persist security info=False; user id=app:appid@tenantid;initial catalog=TabularProject15;data source=powerbi://api.powerbi.com/v1.0/myorg/Premium%20Test;update isolation level=2;Password=<application secret>

![New Providerjpg](https://github.com/TabularEditor/TabularEditor3/assets/144365095/afbb6708-ba9a-4ab4-a948-5ede3e89bfa3)

My data link properties provider is:

![data link properties Provider](https://github.com/TabularEditor/TabularEditor3/assets/144365095/d2d7ada2-91e2-4834-aff9-eae7aee02f34)

When I click test connection I got this

![test](https://github.com/TabularEditor/TabularEditor3/assets/144365095/97c9225a-f93f-4bbd-b0c2-66262d69c895)

When I continue with the Test Connection I get this

![test 2](https://github.com/TabularEditor/TabularEditor3/assets/144365095/bcc08e9f-8115-4643-9d0d-174575968e7b)

What I am doing wrong that I get that Multiple-step OLE DB operation generated errors message?

Thanks

Richard

**Comment by otykier** [comment, marked as answer]
Hi Richard.
I'm not sure why the multiple-step error is showing - I get the same thing on my end. We're just using the standard API for the Data Links dialog, so anything that goes on inside that dialog is unfortunately out of our hands. However, the good news is that you can still import tables and use TE3's update table schema feature after setting up the legacy data source this way. Refresh operations also seem to work fine. So I think it's safe to ignore that error message.

**Comment by rhains** [comment]
Hi Daniel, 

Would you be able to tell me, once you setup the Data source being a Tabular Model, and after you deployed into a Power BI Premium Workspace, how do you make Power Bi to recognize the data source?

 

Or, may be if you can send me an example of the Connection String you use so I can compare with the one I am using. 

 

I will really appreciate.

 

Thanks

 

Richard Hains
